alternator/serialization.cc

```diff
@@ -413,8 +439,9 @@ position_in_partition pos_from_json(const rjson::value& item, schema_ptr schema)
         } else {
             throw std::runtime_error(fmt::format("Invalid value for weight: {}", weight_view));
         }
-        return position_in_partition(region, weight, region == partition_region::clustered ? std::optional(std::move(ck)) : std::nullopt);
+        return position_in_partition(region, weight, region == partition_region::clustered ? std::optional(ck_prefix_from_json(item, schema)) : std::nullopt);
     }
+    auto ck = ck_from_json(item, schema);
```

Comment:
It's very unclear to me when do we ever reach this line. I see this is code which @denesb wrote in commit 2b0bc11f2e5a0d28e8461859186e17bc77a988b8, but I don't know what it does.

Reply:
This code is here to support breaking pages on non-row positions, i.e. range tombstone changes.
This was added as part of the support to break pages at arbitrary positions, in the middle of a tombstone run.
Unlike rows, which have neat positions, expressible with a single clustering key, range tombstone changes are more messy: they are always before/after some row or prefix. Hence the need for `position_in_partition`, which can express such positions. Paging now uses this instead of the plain `clustering_key` used in the simpler days of old.